### PR TITLE
More robust create relative symlink in cache

### PR DESCRIPTION
**Pull request by Wauplin**
Should fix https://github.com/huggingface/huggingface_hub/issues/1108.

### CODE REVIEW

File: src/huggingface_hub/file_download.py [hunk 1]
```diff
@@ -797,15 +797,27 @@ def _create_relative_symlink(src: str, dst: str, new_blob: bool = False) -> None
...
 
     cache_dir = os.path.dirname(os.path.commonpath([src, dst]))
     if are_symlinks_supported(cache_dir=cache_dir):
-        os.symlink(relative_src, dst)
+        relative_src = os.path.relpath(src, start=os.path.dirname(dst))
+        try:
+            os.symlink(relative_src, dst)
+        except FileExistsError:
+            if os.path.islink(dst) and os.path.realpath(dst) == os.path.realpath(src):
+                # `dst` already exists and is a symlink to the `src`` blob. It is most
```
Comment (sgugger) [inline src/huggingface_hub/file_download.py:812]:
```suggestion
                # `dst` already exists and is a symlink to the `src` blob. It is most
```

File: src/huggingface_hub/file_download.py [hunk 2]
```diff
@@ -797,15 +797,27 @@ def _create_relative_symlink(src: str, dst: str, new_blob: bool = False) -> None
...
+                # likely that the file has been cached twice concurrently (exactly
+                # between `os.remove`` and `os.symlink``). Do nothing.
```
Comment (sgugger) [inline src/huggingface_hub/file_download.py:814]:
```suggestion
                # between `os.remove` and `os.symlink`). Do nothing.
```

**Review by sgugger — APPROVED** [review]
Thanks a lot for fixing!

**Comment by Wauplin** [comment]
Cool, thanks for reviewing :)

**Pull request merged**